airbyte-cdk/python/docs/concepts/basic-concepts.md

```diff
@@ -0,0 +1,62 @@
+# Basic Concepts
+
+### The Airbyte Specification
+As a quick recap, the Airbyte Specification requires an Airbyte Source to support 4 distinct operations:
+
+1. `Spec` - The required configuration in order to interact with the underlying technical system e.g. database
+   information, authentication information etc.
+2. `Check` - Validate that the provided configuration is valid with sufficient permissions for one to perform all
+   required operations on the Source.
+3. `Discover` - Discover the Source's schema. This let users select what a subset of the data to sync. Useful
+   if users require only a subset of the data.
+4. `Read` - Perform the actual syncing process. Data is read from the Source, parsed into `AirbyteRecordMessage`s
+   and sent to the Airbyte Destination. Depending on how the Source is implemented, this sync can be incremental
+   or a full-refresh.
+
+A core concept discussed here is the __Source__.
+
+The Source contains one or more __Streams__ (or __Airbyte Streams__). A __Stream__ is the other concept key to
+understanding how Airbyte models the data syncing process. A __Stream__ models the logical data groups that make
+up the larger __Source__. If the __Source__ is a RDMS, each __Stream__ is a table. In a REST API setting, each __Stream__ corresponds
+to one resource within the API. e.g. a __Stripe Source__ would have have one __Stream__ for `Transactions`, one
+for `Charges` and so on.
+
```

Comment:
```suggestion

### The `Source` class

Airbyte provides abstract base classes which make it much easier to perform certain categories of tasks e.g: `HttpStream` makes it easy to create HTTP API-based streams. However, if those do not satisfy your use case (for example, if you're pulling data from a relational database), you can always directly implement the Airbyte Protocol by subclassing the CDK's `Source` class. 

Note that while this is the most flexible way to implement a source connector, it is also the most toilsome as you will be required to manually manage state, input validation, correctly conforming to the Airbyte Protocol message formats, and more. We recommend using a subclass of `Source` unless you cannot fulfill your use case otherwise. 
```